hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/MapNearCacheInvalidationFromClientTest.java

```diff
@@ -241,6 +226,31 @@ public void run() {
         });
     }
 
+    @Test
+    public void testWithTransactionalMap() {
+        TransactionContext context = client.newTransactionContext();
+        context.beginTransaction();
+        try {
+            TransactionalMap<Object, Object> txnMap = context.getMap(mapName);
+            assertNull("Expected null for a non-existent key", txnMap.get("key"));
```

Comment:
These calls are passed as `Data` key to the `TransactionalMapProxy`, which broke the Near Cache lookup before (key was not deserialized).